src/main/java/io/github/syst3ms/skriptparser/expressions/ExprRandomNumber.java

```diff
@@ -0,0 +1,56 @@
+package io.github.syst3ms.skriptparser.expressions;
+
+import io.github.syst3ms.skriptparser.Main;
+import io.github.syst3ms.skriptparser.lang.Expression;
+import io.github.syst3ms.skriptparser.lang.TriggerContext;
+import io.github.syst3ms.skriptparser.parsing.ParseContext;
+import org.jetbrains.annotations.Nullable;
+import java.util.concurrent.ThreadLocalRandom;
+import java.lang.Math;
+/**
+ * Generate a random number (double) or integer.
+ *
+ * @name random number
+ * @pattern [a] random (integer|number) (from|between) %number% (to|and) %number%
+ * @since ALPHA
+ * @author WeeskyBDW
+ */
+public class ExprRandomNumber implements Expression<Number> {
+
+    private Expression<Number> lowerNumber, maxNumber;
+
+	private boolean isInteger;
+    static {
+        Main.getMainRegistration().addExpression(
+                ExprRandomNumber.class,
+                Number.class,
+                true,
+                "[a] random (1:integer|2:number) (from|between) %number% (to|and) %number%"
```

Comment:
```suggestion
                "[a] random integer (from|between) %integer% (to|and) %integer%",
                "[a] random number (from|between) %number% (to|and) %number%"
```
All of my proposed changes are made to avoid having to round the number in the integer case, because that is quite ugly.